src/App.js

```diff
@@ -1,68 +1,84 @@
+import LoadingScreen from "./components/screens/LoadingScreen.js";
+import { useSelector } from 'react-redux';
+
+const AdminRoute = ({ element }) => {
+  const role = useSelector((state) => state.userData.role);
+  console.log("role is "+ role);
+
+  if (role !== 0) {
+    //Redirect to login or unauthorized page if user is not an admin
+    return <Navigate to="/" />;
+  }
+  return element;
+};
 
 const App = () => {
   return (
     <>
       <BrowserRouter>
         <Header />
         <Chatbot
-          title='Mapping of Hit College'
-          src={'temp'}
-          width='100%'
-          height='100%'
-          loading='lazy'
+          title="Mapping of Hit College"
+          src={"temp"}
+          width="100%"
+          height="100%"
+          loading="lazy"
         />
         <Routes>
-          <Route path='/' element={<MainPage />} />
-          <Route path='/power' element={<PowerPage />} />
-          <Route path='/air' element={<AirPage />} />
-          <Route path='/water' element={<WaterPage />} />
+          <Route path="/" element={<MainPage />} />
+          <Route path="/power" element={<PowerPage />} />
+          <Route path="/air" element={<AirPage />} />
+          <Route path="/water" element={<WaterPage />} />
           <Route
-            path='/RecyclingGarbageEfficiency'
+            path="/RecyclingGarbageEfficiency"
             element={<RecyclingPage />}
           />
-          <Route path='/microClimatic' element={<MicroClimaticPage />} />
-          <Route path='/cleanliness' element={<CleanlinessPage />} />
-          <Route path='/students' element={<Students />} />
-          <Route path='/contact' element={<ContactUs />} />
-          <Route path='/staff' element={<Staff />} />
-          <Route path='/news' element={<NewsPage />} />
-          <Route path='/login' element={<Login />} />
-          <Route path='/sign-up' element={<Register />} />
-          <Route path='/airDashboard' element={<AirDashboard />} />
-          <Route path='/powerDashboard' element={<PowerDashboard />} />
-          <Route path='/admin' element={<AdminDashboard />} /> {/* נתיב למסך האדמין */}
-          <Route path='/UsersTable' element={<AdminUsersTable />} />
-          <Route path='/SensorStatistics' element={<SensorStatistics/>} />
-          <Route path='/AlertCenter' element={<AdminAlertCenter/>} />
+          <Route path="/microClimatic" element={<MicroClimaticPage />} />
+          <Route path="/cleanliness" element={<CleanlinessPage />} />
+          <Route path="/students" element={<Students />} />
+          <Route path="/contact" element={<ContactUs />} />
+          <Route path="/staff" element={<Staff />} />
+          <Route path="/news" element={<NewsPage />} />
+          <Route path="/login" element={<Login />} />
+          <Route path="/sign-up" element={<Register />} />
+          <Route path="/airDashboard" element={<AdminRoute element={<AirDashboard />} />} />
```

Comment:
all users can access the dashboards, not just admins. you can create a similar custom route to check if a user is logged in for these